### DISCUSSION TITLE
Why do we need "SpecFlow.Internal.Json"?

### DISCUSSION THREAD

**Discussion by gasparnagy** [reqnroll/Reqnroll#32] [Contributor Q&A] [answered]
@SabotageAndi another question... sry.

I remember we discussed about this, but I don't remember all the details. I know that originally we used Json.NET, but that was conflicting with the version of Json.NET the projects themselfs were using. 

But I don't remember why we cannot use `System.Text.Json` instead. As far as I see, our JSON deserialization needs are very simple, we only need it for loading the config file. (Maybe that was not available at the time we introduced `SpecFlow.Internal.Json`?)

Could you please help with some background info?

**Comment by SabotageAndi** [comment, marked as answer]
Assembly loading again.

As the code behind generation runs in the msbuild process, we are need to use the same versions as msbuild/dotnet.

Which brings problems, when users want to use a higher version of the libraries.
We had this problem with Json.NET.

We switched to Utf8Json because of that.
But this had issues with mono on Mac.

So we switched to our "own" Jsonparser. Which is a fork of https://github.com/zanders3/json packaged into a NuGet package.
We needed this parser also in the closed source parts so we made a NuGet package.

**Reply by SabotageAndi** [reply]
No idea why not System.Text.Json. Could be because of .NET Framework Support

**Reply by ajeckmans** [reply]
System.Text.Json works for .NET framework as well. We switched our old ASP.NET 4.7 applications to System.Text.Json mainly due to shared (internal) packages and the well known issues with Json.NET

**Reply by SabotageAndi** [reply]
Good to know.
Probably we just didn't want to add a third party dependency, because we had so much trouble always with them.

**Comment by gasparnagy** [comment]
OK/thx. Then maybe it would make sense to give a chance to System.Text.Json, but low prio.

**Reply by kwlin** [reply]
@gasparnagy does this issue has a higher prio now since Specflow official is EoL? 

https://github.com/[SpecFlowOSS/SpecFlow](https://github.com/SpecFlowOSS/SpecFlow?tab=readme-ov-file#specflow-eol)?tab=readme-ov-file#specflow-eol

**Reply by gasparnagy** [reply]
Yes, but not immediately. We have the source code of that component, if needed and I don't think it will disappear from NuGet, but you are right, we should get rid o that as soon as possible.

**Reply by obligaron** [reply]
I opened https://github.com/reqnroll/Reqnroll/pull/373 to replace Specflow.Internal.Json with System.Text.Json.
